executor/aggfuncs/func_first_row.go

```diff
@@ -475,25 +502,26 @@ func (e *firstRow4Enum) ResetPartialResult(pr PartialResult) {
 func (e *firstRow4Enum) UpdatePartialResult(sctx sessionctx.Context, rowsInGroup []chunk.Row, pr PartialResult) (memDelta int64, err error) {
 	p := (*partialResult4FirstRowEnum)(pr)
 	if p.gotFirstRow {
-		return 0, nil
+		return memDelta, nil
 	}
 	for _, row := range rowsInGroup {
 		d, err := e.args[0].Eval(row)
 		if err != nil {
-			return 0, err
+			return memDelta, err
 		}
 		p.gotFirstRow, p.isNull, p.val = true, d.IsNull(), d.GetMysqlEnum().Copy()
+		memDelta += int64(d.Length())
```

Comment:
I think the memDelta is the part of string named Name in the set type only, d.Length include the part of uint64 named Value which had calculated in the appedPartialResult.